pufferlib/ocean/trash_pickup/trash_pickup.h

```diff
@@ -0,0 +1,596 @@
+    for (int i = 0; i < env->num_agents; i++){
+         env->rewards[i] += reward;
+    }
+
+    env->total_episode_reward += reward;
+}
+
+void move_agent(CTrashPickupEnv* env, int agent_idx, int action) {
+    int x = env->agent_positions[agent_idx][0];
+    int y = env->agent_positions[agent_idx][1];
+    int carrying = env->agent_carrying[agent_idx];
+
+    int move_dir_x = 0;
+    int move_dir_y = 0;
+    if (action == ACTION_UP) move_dir_y = -1;
+    else if (action == ACTION_DOWN) move_dir_y = 1;
+    else if (action == ACTION_LEFT) move_dir_x = -1;
+    else if (action == ACTION_RIGHT) move_dir_x = 1;
+    else printf("Undefined action: %d", action);
+
+    int new_x = x + move_dir_x;
+    int new_y = y + move_dir_y;
+
+    if (new_x >= 0 && new_x < env->grid_size && new_y >= 0 && new_y < env->grid_size)
+    {
+        int cell_state_type = env->grid[INDEX(env, new_x, new_y)].type;
+        if (cell_state_type == EMPTY) 
+        {
+            env->agent_positions[agent_idx][0] = new_x;
+            env->agent_positions[agent_idx][1] = new_y;
+            env->grid[INDEX(env, x, y)].type = EMPTY;
+            env->grid[INDEX(env, x, y)].index = -1;
+            env->grid[INDEX(env, new_x, new_y)].type = AGENT;
+            env->grid[INDEX(env, new_x, new_y)].index = agent_idx;
+        } 
+        else if (cell_state_type == TRASH && carrying == AGENT_IS_NOT_CARRYING) 
+        {
+            int trash_index = env->grid[INDEX(env, new_x, new_y)].index;
+            env->trash_presence[trash_index] = 0; // Mark as not present
+            env->trash_positions[trash_index][0] = -1;
+            env->trash_positions[trash_index][1] = -1;
+
+            env->agent_carrying[agent_idx] = AGENT_IS_CARRYING;
+            env->grid[INDEX(env, x, y)].type = EMPTY;
+            env->grid[INDEX(env, x, y)].index = -1;
+            env->grid[INDEX(env, new_x, new_y)].type = AGENT;
+            env->grid[INDEX(env, new_x, new_y)].index = agent_idx;
+            env->agent_positions[agent_idx][0] = new_x;
+            env->agent_positions[agent_idx][1] = new_y;
+            add_reward(env, env->positive_reward);
+        } 
+        else if (cell_state_type == TRASH_BIN) 
+        {
+            if (carrying == AGENT_IS_CARRYING)
+            {
+                // Deposit trash into bin
+                env->agent_carrying[agent_idx] = AGENT_IS_NOT_CARRYING;
+                add_reward(env, env->positive_reward);
+            }
+            else
+            {
+                bool try_pull_bin = false;
+                int new_bin_x = new_x + move_dir_x;
+                int new_bin_y = new_y + move_dir_y;
+                if (new_bin_x >= 0 && new_bin_x < env->grid_size && new_bin_y >= 0 && new_bin_y < env->grid_size)
+                {
+                    int new_bin_cell_state_type = env->grid[INDEX(env, new_bin_x, new_bin_y)].type;
+                    if (new_bin_cell_state_type == EMPTY)
```

Comment:
This deep = :/ unless you are implementing a conv

Reply:
Fixed (to be pushed).